Improve progress plugin docs (#3520)
webpack/webpack.js.org: src/content/plugins/progress-plugin.md

When providing an `object` to the `ProgressPlugin`, following properties are supported:

- `activeModules` (`boolean = false`): Shows active modules count and one active module in progress message.
- `entries` (`boolean = true`): Shows entries count in progress message.

> - `handler` (See [Providing function](#providing-function)): `modules` (`boolean = true`): Shows modules count in progress message.

Reviewer: Probably missed a line break here?
```suggestion
- `handler` (See [Providing function](#providing-function))
- `modules` (`boolean = true`): Shows modules count in progress message.
```

- `modulesCount` (`number = 5000`): A minimum modules count to start with. Takes effect when `modules` property is enabled.
- `profile` (`boolean = false`): Tells `ProgressPlugin` to collect profile data for progress steps.

> - `dependencies` (`boolean = true`): Shows dependencies count in progress message.

Reviewer: Is this the right way to say it or do i mix it up?
```suggestion
- `dependencies` (`boolean = true`): Shows the count of dependencies in progress message.
```

- `dependenciesCount` (`number = 10000`): A minimum dependencies count to start with. Takes effect when `dependencies` property is enabled.
- `percentBy` (`string = null: 'entries' | 'dependencies' | 'modules' | null`): Tells `ProgressPlugin` how to calculate progress percentage.

```js
new webpack.ProgressPlugin({
  activeModules: false,
  entries: true,
  handler(percentage, message, ...args) {
    // custom logic
  },
  modules: true,
  modulesCount: 5000,
  profile: false,
  dependencies: true,
  dependenciesCount: 10000,
  percentBy: null
});
```

> > On the projects with many entry points better use `percentBy: 'entries'` because of entry points amount is known in advance, therefore the percentage will be more realistic.

Reviewer: ```suggestion
> We recommend using `percentBy: 'entries'` setting for projects with [multiple configured entry points](/configuration/entry-context/#entry). Percentage calculation will become more accurate because the amount of entry points is known in advance.
```

When providing an `object` to the `ProgressPlugin`, following properties are supported:

> 

Reviewer: this list was a good summary of properties, if you wanted to make it more accessible you could aggregate defaults into this list. Spreading it out the page and copying over the same info just makes it different and not necessarily improve the documentation

Author: I followed this guide https://webpack.js.org/contribute/writers-guide/#configuration-defaults-and-types
All the options are listed in the example ahead
Do we really need to revert this changes?

Reviewer: they are usually done per config property, since config property itself is object its fine to keep it as a list, what i meant is adding types and defaults (as you did) but to this list to keep it as a short summary (as info did not really expand much)

for example 
```diff
- - `activeModules` show's active modules ...
+ - `activeModules` (`boolean = false`): show's active modules ...
```

Like we did previously here: https://webpack.js.org/plugins/eval-source-map-dev-tool-plugin/#options

```js
new webpack.ProgressPlugin({
  activeModules: false,
  entries: true,
  handler(percentage, message, ...args) {
    // custom logic
  },
  modules: true,
  modulesCount: 5000,
  profile: false,
  dependencies: true,
  dependenciesCount: 10000,
  percentBy: null
});
```

#### `activeModules`

`boolean = false`

Shows active modules count and one active module in progress message.

#### `entries`

`boolean = true`

Shows entries count in progress message.

#### `handler`

> See [Providing function](#providing-function)

Reviewer: this doesnt look right in an h4 formatted list

#### `modules`

`boolean = true`

Shows modules count in progress message.

#### `modulesCount`

`number = 5000`

A minimum modules count to start with. Takes effect when `modules` property is enabled.

#### `profile`

`boolean = false`

Tells `ProgressPlugin` to collect profile data for progress steps.

> - `percentBy` tells `ProgressPlugin` how to calculate progress percentage.

Reviewer: ?

#### `dependencies`

`boolean = true`

Shows dependencies count in progress message.

#### `dependenciesCount`

`number = 10000`

A minimum dependencies count to start with. Takes effect when `dependencies` property is enabled.

#### `percentBy`

`string = null: 'entries' | 'dependencies' | 'modules' | null`

Tells `ProgressPlugin` how to calculate progress percentage.

> On the projects with tons of entry points better use `percentBy: 'entries'` because the amount of entry points is known in advance, therefore the percentage will be more realistic.

Reviewer: "tons of entries" is not of a documentation level. 
therefore the percentage will be more realistic - will be more accurate sounds better
